### PR TITLE
Separate common code for remote target FW upgrade supported optics

### PR DESCRIPTION
**Pull request by mihirpat1**
#### Description

We need to separate the common code for optics supporting target side FW upgrade.
Currently, the following function needs to be moved to a common file and class representing remote target FW upgradable modules.
https://github.com/sonic-net/sonic-platform-common/blob/2606cd5fb0a58fc956577602235a9f8f84a887c6/sonic_platform_base/sonic_xcvr/api/credo/aec_800g.py#L10-L11
MSFT ADO - 27595597

#### Motivation and Context

With the introduction of different types of CMIS modules supporting remote target FW upgrade, a common driver needs to be implemented to reduce replication of code across multiple optic vendors.
If any vendor specific optic requires a specific implementation, the corresponding vendor should override the relevant APIs of `CmisTargetFWUpgradeAPI`.

#### How Has This Been Tested?

The change set has been tested on a module supporting remote target FW upgrade. Following were the testcases executed
```
1. Output of  sfputil show fwversion -p PORT_NAME
2. sfputil firmware download PORT_NAME -d <>
3. sfputil firmware run PORT_NAME
4. sfputil firmware commit PORT_NAME
```

### CODE REVIEW

File: sonic_platform_base/sonic_xcvr/mem_maps/public/cmisTargetFWUpgrade.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    NumberRegField,
+    RegGroupField,
+)
+from ...fields import consts
+
+class CmisTargetFWUpgradeMemMap(CmisMemMap):
+    def __init__(self, codes):
+        super().__init__(codes)
+
+        self.VENDOR_CUSTOM = RegGroupField(consts.VENDOR_CUSTOM,
```
Comment (prgeor) [inline sonic_platform_base/sonic_xcvr/mem_maps/public/cmisTargetFWUpgrade.py:19]: @mihirpat1 can we keep this in Credo specific memmap?

Comment (mihirpat1) [reply sonic_platform_base/sonic_xcvr/mem_maps/public/cmisTargetFWUpgrade.py]: @prgeor I have addressed this now.

**Review by prgeor — APPROVED** [review]

**Comment by mihirpat1** [comment]
@StormLiangMS @yxieca It will be great if you can help to cherry-pick this to 202305 and 202311.
MSFT ADO - 27595597

**Comment by StormLiangMS** [comment]
hi @mihirpat1  is this a must to have for 202305? If so, for which platform?

**Comment by mihirpat1** [comment]
> hi @mihirpat1 is this a must to have for 202305? If so, for which platform?

@StormLiangMS Yes this is a must have for 202305 on Cisco-8111.

**Pull request merged**